mcworldlib/region.py

```diff
@@ -139,18 +138,20 @@ def parse(cls, buff):
             buff.seek(offset)
             chunk = RegionChunk.parse(buff, header=header)
 
-            # TODO: Replace asserts with proper Exceptions and/or logging
-
-            # ~2001-09-09 GMT
-            assert timestamp  > 1000000000, \
-                f'Invalid timestamp for chunk {pos}: {timestamp} ({u.isodate(timestamp)})'
+            # timestamp should be after ~2001-09-09 GMT
+            if timestamp < 1000000000:
+                log.warning(f'Invalid timestamp for chunk {pos}: {timestamp} ({u.isodate(timestamp)})')
 
+            # Warn when sector_count does not match expected as declared in the region header.
             # Sometimes Minecraft saves sector_count + 1 when chunk length
             # (including header) is an exact multiple of SECTOR_BYTES
-            assert chunk.sector_count <= sector_count <= chunk.sector_count + 1, \
-                f'Length mismatch for region {self.pos} in chunk {pos}:' \
-                f' region header declares {sector_count} {SECTOR_BYTES}-byte sectors,' \
-                f' but chunk data required {chunk.sector_count}.'
+            if chunk.sector_count != sector_count:
+                if chunk.sector_count + 1 != sector_count:
+                    log.warning(
+                        f'Length mismatch for region {self.pos} in chunk {pos}:'
+                        f' region header declares {sector_count} {SECTOR_BYTES}-byte sectors,'
+                        f' but chunk data required {chunk.sector_count}.'
+                    )
```

Comment:
Great! It really needed to replace this assert with a proper logging! But why change the original condition  with double nested `if`s? I believe a single `if not(chunk.sector_count <= sector_count <= chunk.sector_count + 1): log.warning(...` will do the trick

Reply:
I believe the main reason was that I wanted to ensure that this warning would be printed for mismatches of any variety, not just if `chunk.sector_count > sector_count`.

My thinking was that you can't chain general inequality together and be guaranteed the expected result in every configuration, but I may be wrong on that.

However, even given that, there are now a couple of conditional flaws I'm seeing here, for example, the previous assertion would have thrown an error if `sector_count == chunk.sector_count + 1` but my code would not.

At the very least, I'll need to fix that much. If you require that the conditional match the original code, then I can make it so.

Reply:
The idea is to allow `sector_count` to be either `chunk.sector_count` or `chunk.sector_count + 1`; any other value will emit a warning. I think `chunk.sector_count <= sector_count <= chunk.sector_count + 1` covers that, but there are my ways to write that (`if sector_count not in (chunk.sector_count, chunk.sector_count+1)` would be another, or using `range()`. Pick the one you think you prefer and think it's most readable. No, the original assertion would not throw error if `sector_count == chunk.sector_count + 1`.